### PR TITLE
47373_supervisor_add_subjects

### PR DESCRIPTION
**Pull request by asifbuetcse**

### CODE REVIEW

File: resources/views/supervisor/tasks/create.blade.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                        {!! Form::textarea('description') !!}
+                    </div>
+                    <div class="form-group">
+                        <div class="col-sm-offset-3 col-sm-6">
+                            {!! Form::submit('Add Task') !!}
+                        </div>
+                         {!! link_to_route('supervisor.subjects.index', 'finish') !!}
+                    </div>
+                {!! Form::close() !!}
+
```
Comment (Omid07) [inline resources/views/supervisor/tasks/create.blade.php]: remove this blank line

**Comment by Omid07** [comment]
ok

File: app/Http/Controllers/Supervisor/TasksController.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        return view('supervisor.tasks.create', compact('subject'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
```
Comment (yasin-buet) [inline app/Http/Controllers/Supervisor/TasksController.php]: please use validator for checking the incoming request

Comment (asifbuetcse) [inline app/Http/Controllers/Supervisor/TasksController.php]: done.

**Comment by yasin-buet** [comment]
OK

File: app/Http/Controllers/Supervisor/TasksController.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+     * @return \Illuminate\Http\Response
+     */
+    public function create(Request $request)
+    {
+        if ($request->session()->has('subjectId')) {
+            $subjectId = \Session::get('subjectId');
+            $subject = Subject::find($subjectId);
+        }
+        else {
+            $subject = Subject::all()->last();
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Supervisor/TasksController.php]: Route should be `supervisor/subjects/$subjectId/tasks/create` 
Please create new controller. `SubjectTasksController`
http://laravel.com/docs/5.1/controllers#restful-nested-resources

File: app/Http/routes.php [hunk 4]
```diff
@@ -32,4 +32,5 @@
     Route::resource('courses', 'CoursesController');
     Route::resource('subjects', 'SubjectsController');
     Route::resource('users', 'UsersController');
+    Route::resource('subjetcs.tasks', 'SubjectsTasksController');
```
Comment (dinhhoanglong91) [inline app/Http/routes.php]: type error `subjects`

File: app/Http/Controllers/Supervisor/SubjectsController.php [hunk 5]
```diff
@@ -40,7 +40,13 @@ public function create()
...
     public function store(Request $request)
     {
-        //
+        $this->validate($request, [
+        'name' => 'required|max:255',
+        'description' => 'required',
+        ]);
+        $input = $request->all();
+        $subjectId = Subject::create($input)->id;
+        return redirect()->action('Supervisor\SubjectsTasksController@create')->with('subjectId', $subjectId);
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Supervisor/SubjectsController.php]: https://github.com/framgia/tms_42/pull/16#discussion_r46929546
If subject id is 1, route should be `subjects/1/tasks/create`.
Task belongs to subject and task id is detected by router, you don't have to check subject id in session.

File: app/Http/Controllers/Supervisor/SubjectsTasksController.php [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        return view('supervisor.tasks.create', compact('subject'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Supervisor/SubjectsTasksController.php]: Route is: `subjects/1/tasks/store`. I wonder if you can get `subject_id` from this route?
Please try
```php
public function store(Request $request, $subjectId)
```

**Comment by dinhhoanglong91** [comment]
One subject has multiple tasks, but in your pull request, you can add only one tasks, right?
If you can add multiple tasks in one screen, there is no problem. And you have to fix your pull request. Otherwise, you have to change follow
1. Admin add subject `/subjects/add`
2. Admin view subject detail `/subjects/1`
There is list of tasks in this page. Firstly, there is no task
3. There is a button to "Add task". Link to `/subjects/1/tasks/create`
4. After admin adds task, go back to subject detail screen to see subject name, tasks list.

File: app/Http/Controllers/Supervisor/SubjectsTasksController.php [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        return view('supervisor.tasks.create', compact('subject'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request, $id)
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Supervisor/SubjectsTasksController.php]: Change `$id` to `$subjectId`

File: app/Http/Controllers/Supervisor/SubjectsTasksController.php [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    {
+        //
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create($id)
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Supervisor/SubjectsTasksController.php]: Change `$id` to `$subjectId`

**Comment by dinhhoanglong91** [comment]
ok

**Pull request merged**